spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestMerge.java

```diff
@@ -523,6 +523,78 @@ public void testMergeWithOnlyUpdateClauseAndNullValues() {
         sql("SELECT * FROM %s ORDER BY id", selectTarget()));
   }
 
+  @Test
+  public void testMergeWithOnlyUpdateNullUnmatchedValues() {
```

Comment:
What about adding these tests to all Spark version to be safe?

Reply:
Great point, yes we should. Done!